### PR TITLE
add Idle v3

### PR DESCRIPTION
**Pull request by krzysu**
Hi, I added Idle's latest contracts to money-legos.
Their docs https://developers.idle.finance/ and website https://idle.finance/#/

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -28,7 +28,9 @@
 - DappSys
 - DyDx
 - ERC20
+- Idle V3
+- Kyber.Network
 - MakerDAO
 - OneSplit
+- UMA
```
Comment (adrianmcli) [inline docs/README.md:35]: thanks for adding these!

File: docs/overview_mainnet_address.md [hunk 2]
```diff
@@ -105,6 +105,24 @@
  - weth: [0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2](https://etherscan.io/address/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2)
  - zrx: [0xE41d2489571d322189246DaFA5ebDe1F4699F498](https://etherscan.io/address/0xE41d2489571d322189246DaFA5ebDe1F4699F498)
 
+## Idle
```
Comment (adrianmcli) [inline docs/overview_mainnet_address.md:108]: Unfortunately, I think your addresses in the `overview_mainnet_address.md` will get overwritten unless you write the script to insert the Idle Finance addresses. See below.

See docs build script here: https://github.com/studydefi/money-legos/blob/master/package.json#L13

File: scripts/genAddrMainnetOverview.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const formatToEtherscanLink = (l) =>
 
 // Funky protocols with different structures
 // Need to construct the markdown a bit differently
-const funkyProtocols = ["curvefi"];
+const funkyProtocols = ["curvefi", "idle"];
```
Comment (adrianmcli) [inline scripts/genAddrMainnetOverview.js:10]: You'll need to write a section here where you call `markdown.push` on the addresses you want to see show up. We should probably refactor this to be a bit better, but this is what we have right now.

**Comment by adrianmcli** [comment]
You can test building the docs with `yarn build:docs`. We probably need to have that in our CI actually.

File: scripts/genAddrMainnetOverview.js (hunk 3 above)
Comment (krzysu) [reply scripts/genAddrMainnetOverview.js:10]: in fact, I did use `yarn build:docs` to generate `overview_mainnet_address.md`. As my `contracts` object is deeper than one level, I found that the script for "curvefi" also works for "idle". 
That is why I added it to `funkyProtocols`. Try to run `yarn build:docs` yourself 😃

File: docs/overview_mainnet_address.md (hunk 2 above)
Comment (krzysu) [reply docs/overview_mainnet_address.md:108]: this was generated from your script 😄

**Comment by adrianmcli** [comment]
Oh strange, I just tried to run the script and it didn't overwrite the addresses. Maybe my understanding of the script is wrong.

File: docs/overview_mainnet_address.md (hunk 2 above)
Comment (adrianmcli) [reply docs/overview_mainnet_address.md:108]: nevermind! good work, my bad.

File: scripts/genAddrMainnetOverview.js (hunk 3 above)
Comment (adrianmcli) [reply scripts/genAddrMainnetOverview.js:10]: I just did, thank you for the explanation!

**Pull request merged**